src/hubbleds/pages/06-prodata/__init__.py

```diff
@@ -0,0 +1,208 @@
+# Stage 6 (prodata)
+import solara
+
+from cosmicds.components import ScaffoldAlert
+from cosmicds import load_custom_vue_components
+
+from cosmicds.components import LayerToggle
+# from cosmicds.components import ViewerLayout
+from ...components import GridViewer
+from ...viewers import HubbleFitView
+
+from glue_jupyter.app import JupyterApplication
+
+
+from reacton import ipyvuetify as rv
+from solara import Reactive
+
+from pathlib import Path
+
+from ...data_management import *
+from ...state import GLOBAL_STATE, LOCAL_STATE, mc_callback
+# import for type definitions
+from typing import cast
+
+from .component_state import ComponentState, Marker
+
+# the guidelines in the current files parent directory
+GUIDELINE_ROOT = Path(__file__).parent / "guidelines"
+
+# intitialize the component state
+component_state = ComponentState()
+
+
+
+def basic_viewer_setup(viewer_class, glue_session, data_collection, name, x_att, y_att):
+    viewer = viewer_class(glue_session)
+    viewer.add_data(data_collection[name])
+    viewer.state.x_att = x_att
+    viewer.state.y_att = y_att
+    return viewer
+
+    
+# create the Page for the current stage
+@solara.component
+def Page():
+    
+    # NM: Custom vue-only components have to be registered in the Page element
+    #  currently, otherwise they will not be available in the front-end 
+    load_custom_vue_components() # currently this just loads the ScaffoldAlert component
+    
+    # NM: Solara's reactivity is often tied to the _context_ of the Page it's
+    #  being rendered in. Currently, in order to trigger subscribed callbacks,
+    #  state connections need to be initialized _inside_ a Page.
+    component_state.setup()
+    
+    # create glue app with the global data collection and session
+    def glue_setup() -> JupyterApplication:
+        gjapp = JupyterApplication(GLOBAL_STATE.data_collection, GLOBAL_STATE.session)
+        viewer = gjapp.new_data_viewer(HubbleFitView, show=False)
+        viewer.state.title = "Professional Data"
+        viewer.figure.update_xaxes(showline=True, mirror=False)
+        viewer.figure.update_yaxes(showline=True, mirror=False)
+        return gjapp, viewer
+    gjapp, viewer = cast(JupyterApplication, solara.use_memo(glue_setup,[]))
+    
+    #TODO: Should viewer creation happen somewhere else?
+    
+    # viewer.toolbar.set_tool_enabled("hubble:linefit", False)
+    component_state.add_data_by_marker(viewer)
+    component_state.show_legend(viewer, show=True)
+    
+    mc_scoring, set_mc_scoring  = solara.use_state(LOCAL_STATE.mc_scoring.value)
+        
+
+    
+    solara.Markdown(
+        f"""
+        Current Step: {component_state.current_step.value}
+
+        Next step: {Marker.next(component_state.current_step.value)}
+
+        Can advance: {component_state.can_transition(next=True)}
+
+        mc-scoring: {mc_scoring}
+
+        """
+    )
+    
+    
+    with solara.Row():
+        with solara.Column():
```

Comment:
I found that using `solara.Row` and `solara.Column` resulted in incorrect column widths (they're equal sized instead of 1:2). Is this turning out okay for you? If not, I'd suggest using `rv.Row` and `rv.Col` as in the Spectra & Velocity page.

Reply:
I had not paid attention to the styling. I will update it to use the rv.Col with the appropriate cols=4 and 8 widths

Reply:
I used `solara.ColumnsResponsive` as in [here](https://github.com/patudom/hubbleds/blob/bd4ca86818de818d679ef505b3ba672ae2b23140/src/hubbleds/pages/03-distance-measurements/__init__.py#L99), which ports over how we handled this in the voila version. I found that solara didn't really care if I included Rows or not, so I think I left those out in some places and it behaved correctly.